src/services/metrics.py

```diff
@@ -121,3 +121,8 @@ async def monitor_http_requests(self, req: Request, call_next: Any) -> Any:
 
         # continue request handling.
         return response
+
+    @classmethod
+    def _reset_for_tests(cls) -> None:
+        """Reset the singleton instance. Only use this for testing purpose."""
+        SingletonMeta.reset_instance(cls)
```

Comment:
(nit) Not a good solution, actual code should not contain anything specific for tests. But acceptable for now!

Reply:
I understand your point. I found some recommendations for `reset` functions for singleton testing. But I think this contradicts the idea of a singleton and a regular reset function could cause a lot of trouble. So making it private and naming `...for_test` was my compromise. But yeah, maybe it is still not a good idea.